pettingzoo/butterfly/knights_archers_zombies/knights_archers_zombies.py

```diff
@@ -45,24 +46,43 @@ def __init__(
         spawn_rate=20,
         num_archers=2,
         num_knights=2,
+        max_zombies=15,
+        max_arrows=15,
         killable_knights=True,
         killable_archers=True,
         pad_observation=True,
         line_death=False,
         max_cycles=900,
+        vector_state=False,
+        use_typemasks=True,
     ):
         EzPickle.__init__(
             self,
             spawn_rate,
             num_archers,
             num_knights,
+            max_zombies,
+            max_arrows,
             killable_knights,
             killable_archers,
             pad_observation,
             line_death,
             max_cycles,
+            vector_state,
+            use_typemasks,
         )
 
+        # whether we want RGB state or vector state
+        self.vector_state = vector_state
+        # agents + zombies + weapons
+        self.num_tracked = (
+            num_archers + num_knights + max_zombies + num_knights + max_arrows
+        )
+        self.use_typemasks = use_typemasks
+        self.typemask_width = 6
```

Comment:
Why is it 6? The PR description states it should be 4 if I understand that right.

Reply:
The PR description is outdated, I ought to update it, I'll do that within the next few hours from now. But the reason it's 6 is because we also track the arrows and swords, so it's now [zombie, archer, knight, arrow, sword, you].